exonum/src/storage/proof_list_index/tests.rs

```diff
@@ -769,3 +769,47 @@ mod rocksdb_tests {
         super::same_merkle_root(db1, db2);
     }
 }
+
+mod root_hash_tests {
+    use crypto::{self, Hash};
+    use storage::{Database, MemoryDB, ProofListIndex};
+
+    /// Cross-verify `root_hash()` with `ProofListIndex` against expected root hash value.
+    fn assert_root_hash_correct(hashes: &[Hash]) {
```

Comment:
Nit: I'd suggest moving the most interesting code here — tests — to the top, and utility methods — after their usages (i.e., after tests).

Reply:
Hm... I prefer the current situation: all the logic tests use at the top (so you'll know what's going on), and tests at the bottom (so you are prepared for them).

IMHO it's a bit of holy-war question :)